src/range.cr

```diff
@@ -305,4 +305,46 @@ struct Range(B, E)
       self
     end
   end
+
+  #Perform binary search
+  #Based on Marc-André Lafortune's Ruby backports implementation, ported by T. Yamada
+  def bsearch
+    from = self.begin
+    to   = self.end
+    unless from.is_a?(Int) && to.is_a?(Int)
+      # Float support is currently dropped
+      raise "can't do binary search for #{from.class}"
+    end
+
+    midpoint = 0 # placeholder
+    convert = ->{ midpoint }
+
+    to -= 1 if excludes_end?
+    satisfied = nil
+    while from <= to
+      midpoint = (from + to)/2
+      result = yield(cur = convert.call)
```

Comment:
Could you elaborate what the deal with `convert` is? Seems like useless overhead so far.